File: frontend/lib/features/students/data/models/student.dart
enum StudentStatus {
  active('active'),
  inactive('inactive');

  const StudentStatus(this.apiValue);

  final String apiValue;

  static StudentStatus fromApiValue(String value) => StudentStatus.values.firstWhere((s) => s.apiValue == value);
}

/// The student's current bus route + stop (Phase 14), or absent when they
/// don't use school transport.
class StudentTransport {
  const StudentTransport({
    required this.routeId,
    required this.routeName,
    required this.routeLabel,
    required this.vehicleName,
    required this.stopId,
    required this.stopName,
  });

  factory StudentTransport.fromJson(Map<String, dynamic> json) {
    return StudentTransport(
      routeId: json['route_id'] as int,
      routeName: json['route_name'] as String,
      routeLabel: json['route_label'] as String,
      vehicleName: json['vehicle_name'] as String?,
      stopId: json['stop_id'] as int,
      stopName: json['stop_name'] as String,
    );
  }

  final int routeId;
  final String routeName;

  /// "Bus 04 - Green Park".
  final String routeLabel;
  final String? vehicleName;
  final int stopId;
  final String stopName;
}

class Student {
  const Student({
    required this.id,
    required this.schoolId,
    required this.schoolName,
    required this.classSectionId,
    required this.classSectionName,
    required this.admissionNumber,
    required this.firstName,
    required this.lastName,
    required this.name,
    required this.rollNumber,
    required this.guardianName,
    required this.guardianMobile,
    required this.address,
    required this.status,
    this.transport,
    this.guardianEmail,
    this.studentMobile,
    this.studentEmail,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'] as int,
      schoolId: json['school_id'] as int,
      schoolName: json['school_name'] as String?,
      classSectionId: json['class_section_id'] as int?,
      classSectionName: json['class_section_name'] as String?,
      admissionNumber: json['admission_number'] as String,
      firstName: json['first_name'] as String,
      lastName: json['last_name'] as String,
      name: json['name'] as String,
      rollNumber: json['roll_number'] as String?,
      guardianName: json['guardian_name'] as String,
      guardianMobile: json['guardian_mobile'] as String?,
      address: json['address'] as String?,
      status: StudentStatus.fromApiValue(json['status'] as String),
      transport: json['transport'] == null
          ? null
          : StudentTransport.fromJson(json['transport'] as Map<String, dynamic>),
      guardianEmail: json['guardian_email'] as String?,
      studentMobile: json['student_mobile'] as String?,
      studentEmail: json['student_email'] as String?,
    );
  }

  final int id;
  final int schoolId;
  final String? schoolName;
  final int? classSectionId;
  final String? classSectionName;
  final String admissionNumber;
  final String firstName;
  final String lastName;
  final String name;
  final String? rollNumber;
  final String guardianName;
  final String? guardianMobile;
  final String? address;
  final StudentStatus status;
  final StudentTransport? transport;

  /// Contact details beyond the guardian's mobile - each optional, and each
  /// a place the school can reach the family (see Phase 16, email channel).
  final String? guardianEmail;
  final String? studentMobile;
  final String? studentEmail;

  Student copyWith({StudentStatus? status}) {
    return Student(
      id: id,
      schoolId: schoolId,
      schoolName: schoolName,
      classSectionId: classSectionId,
      classSectionName: classSectionName,
      admissionNumber: admissionNumber,
      firstName: firstName,
      lastName: lastName,
      name: name,
      rollNumber: rollNumber,
      guardianName: guardianName,
      guardianMobile: guardianMobile,
      address: address,
      status: status ?? this.status,
      transport: transport,
      guardianEmail: guardianEmail,
      studentMobile: studentMobile,
      studentEmail: studentEmail,
    );
  }
}

File: backend/app/Models/Student.php
<?php

namespace App\Models;

use App\Enums\StudentStatus;
use Database\Factories\StudentFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Casts\Attribute;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasOne;

#[Fillable([
    'school_id', 'class_section_id', 'admission_number', 'first_name', 'last_name',
    'roll_number', 'guardian_name', 'guardian_mobile', 'address', 'status',
])]
class Student extends Model
{
    /** @use HasFactory<StudentFactory> */
    use HasFactory;

    protected function casts(): array
    {
        return [
            'status' => StudentStatus::class,
        ];
    }

    /**
     * Convenience accessor for display - not a stored column.
     */
    protected function name(): Attribute
    {
        return Attribute::get(fn () => trim("{$this->first_name} {$this->last_name}"));
    }

    /**
     * @return BelongsTo<School, $this>
     */
    public function school(): BelongsTo
    {
        return $this->belongsTo(School::class);
    }

    /**
     * @return BelongsTo<ClassSection, $this>
     */
    public function classSection(): BelongsTo
    {
        return $this->belongsTo(ClassSection::class);
    }

    /**
     * The student's current transport route + stop, if they use school
     * transport (Phase 14).
     *
     * @return HasOne<StudentTransportAssignment, $this>
     */
    public function transportAssignment(): HasOne
    {
        return $this->hasOne(StudentTransportAssignment::class);
    }

    public function isActive(): bool
    {
        return $this->status === StudentStatus::Active;
    }
}
